programs/x509/cert_write.c

```diff
@@ -517,6 +541,90 @@ int main(int argc, char *argv[])
                 *tail = ext_key_usage;
                 tail = &ext_key_usage->next;
 
+                q = r;
+            }
+        } else if (strcmp(p, "san") == 0) {
+            char *subtype_value;
+            prev = NULL;
+
+            while (q != NULL) {
+                char *semicolon;
+                r = q;
+
+                /* Find the first non-escaped ; occurrence and remove escaped ones */
+                do {
+                    if ((semicolon = strchr(r, ';')) != NULL) {
+                        if (*(semicolon-1) != '\\') {
+                            r = semicolon;
+                            break;
+                        }
+                        /* Remove the escape character */
+                        size_t size_left = strlen(semicolon);
+                        memmove(semicolon-1, semicolon, size_left);
+                        *(semicolon + size_left - 1) = '\0';
+                        /* r will now point at the character after the semicolon */
+                        r = semicolon;
```

Comment:
This is O(n²); not sure if that matters in practice.  Also it may be is necessary to support escaping backslashes.